api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ConceptTranslatorImpl.java

```diff
@@ -52,14 +54,21 @@ public CodeableConcept toFhirResource(@Nonnull Concept concept) {
 				continue;
 			}
 			
-			String mappingType = mapping.getConceptMapType().getName();
-			if (mappingType == null || !mappingType.equalsIgnoreCase("SAME-AS")) {
+			ConceptMapType mapType = mapping.getConceptMapType();
+			boolean sameAs = mapType.getUuid() != null && mapType.getUuid().equals(ConceptMapType.SAME_AS_MAP_TYPE_UUID);
+			sameAs = sameAs || (mapType.getName() != null && mapType.getName().equalsIgnoreCase("SAME-AS"));
+			if (!sameAs) {
 				continue;
 			}
 			
 			ConceptReferenceTerm crt = mapping.getConceptReferenceTerm();
 			
 			String sourceUrl = conceptSourceToURL(crt.getConceptSource().getName());
+			if (sourceUrl == null) {
+				if (Duration.SNOMED_CT_CONCEPT_SOURCE_HL7_CODE.equals(crt.getConceptSource().getHl7Code())) {
+					sourceUrl = FhirConstants.SNOMED_SYSTEM_URI;
+				}
+			}
```

Comment:
Since there is an OpenMRS constant for the snomed-ct hl7 code, we should attempt to add a snomed-ct mapping if not otherwise found by url in the mapping table.